crates/core/src/host/wasm_common/module_host_actor.rs

```diff
@@ -258,11 +258,18 @@ impl<T: WasmModule> Module for WasmModuleHostActor<T> {
 
         db.with_read_only(|tx| {
             log::debug!("One-off query: {query}");
-            // NOTE(jgilles): this returns errors about mutating queries as SubscriptionErrors, which is perhaps
-            // mildly confusing, since the user did not subscribe to anything. Should we rename SubscriptionError to ReadOnlyQueryError?
-            let compiled = crate::subscription::query::compile_read_only_query(db, tx, &auth, &query)?;
 
-            sql::execute::execute_sql(db, tx, compiled.into_iter().map(CrudExpr::Query).collect(), auth)
+            let compiled = sql::compiler::compile_sql(db, tx, &query)?
+                .into_iter()
+                .map(|expr| {
+                    if matches!(expr, CrudExpr::Query { .. }) {
+                        Ok(expr)
+                    } else {
+                        Err(anyhow!("One-off queries are not allowed to modify the database"))
```

Comment:
@kazimuth Could you eyeball this? I'm wondering if we actually want to return an error here -- we could also just filter out any non-`SELECT` statements and return empty results for them.